src/flashmint/builders/leveraged-extended.test.ts

```diff
@@ -0,0 +1,252 @@
+    const refTx = await contract.populateTransaction.issueSetFromExactETH(
+      indexToken,
+      buildRequest.outputTokenAmount,
+      buildRequest.swapDataDebtCollateral,
+      buildRequest.swapDataInputOutputToken,
+      buildRequest.priceEstimateInflator,
+      buildRequest.maxDust,
+      { value: buildRequest.inputTokenAmount }
+    )
+    const builder = new LeveragedExtendedTransactionBuilder(provider)
+    const tx = await builder.build(buildRequest)
+    if (!tx) fail()
+    expect(tx.to).toBe(FlashMintLeveragedExtendedAddress)
+    expect(tx.data).toEqual(refTx.data)
+    expect(tx.value).toEqual(buildRequest.inputTokenAmount)
+  })
+
+  test('returns a tx for redeeming ETH2X (ERC20)', async () => {
+    const buildRequest = createBuildRequest(
+      false,
+      IndexCoopEthereum2xIndex.addressArbitrum!,
+      IndexCoopEthereum2xIndex.symbol,
+      usdcAddress,
+      'USDC'
+    )
+    const refTx = await contract.populateTransaction.redeemExactSetForERC20(
+      buildRequest.inputToken,
+      buildRequest.inputTokenAmount,
+      buildRequest.outputToken,
+      buildRequest.outputTokenAmount,
+      buildRequest.swapDataDebtCollateral,
+      buildRequest.swapDataInputOutputToken
+    )
+    const builder = new LeveragedExtendedTransactionBuilder(provider)
+    const tx = await builder.build(buildRequest)
+    if (!tx) fail()
+    expect(tx.to).toBe(FlashMintLeveragedExtendedAddress)
+    expect(tx.data).toEqual(refTx.data)
+  })
+
+  test('returns a tx for redeeming ETH2X (ETH)', async () => {
+    const buildRequest = createBuildRequest(
+      false,
+      IndexCoopEthereum2xIndex.addressArbitrum!,
+      IndexCoopEthereum2xIndex.symbol,
+      eth,
+      'ETH'
+    )
+    const refTx = await contract.populateTransaction.redeemExactSetForETH(
+      buildRequest.inputToken,
+      buildRequest.inputTokenAmount,
+      buildRequest.outputTokenAmount,
+      buildRequest.swapDataDebtCollateral,
+      buildRequest.swapDataInputOutputToken
+    )
+    const builder = new LeveragedExtendedTransactionBuilder(provider)
+    const tx = await builder.build(buildRequest)
+    if (!tx) fail()
+    expect(tx.to).toBe(FlashMintLeveragedExtendedAddress)
+    expect(tx.data).toEqual(refTx.data)
+  })
+})
+
+function createBuildRequest(
+  isMinting = true,
+  inputToken: string = usdcAddress,
+  inputTokenSymbol = 'USDC',
+  outputToken: string = IndexCoopEthereum2xIndex.addressArbitrum!,
+  outputTokenSymbol: string = IndexCoopEthereum2xIndex.symbol
+): FlashMintLeveragedExtendedBuildRequest {
+  return {
+    isMinting,
+    inputToken,
+    inputTokenSymbol,
+    outputToken,
+    outputTokenSymbol,
+    inputTokenAmount: wei(1),
+    outputTokenAmount: BigNumber.from(194235680),
+    swapDataDebtCollateral: isMinting
+      ? collateralDebtSwapData['icETH']
+      : debtCollateralSwapData['icETH'],
+    swapDataInputOutputToken: isMinting
+      ? inputSwapData['icETH']['ETH']
+      : outputSwapData['icETH']['ETH'],
```

Comment:
This was quickly copied both swap data tuples above could be updated to something more realistic.